### ISSUE TITLE
Lint Request: Detect when a Future is being interpolated into a string

### ISSUE THREAD

**Issue by shyndman** [dart-lang/linter#2048]
**Describe the rule you'd like to see implemented**
I'd like to be warned if a Future is being interpolated into a string. In my experience it's almost always unintended.

I see this crop up quite a bit when migrating between synchronous and asynchronous APIs.

**(Contrived) example**

The following would trigger the lint, as `contents` is a `Future<String>`, not a `String` as the developer intended:

```dart
String readFileContents(String path) {
  final contents = File(path).readAsString();
  return '''
    $path:
    $contents
  ''';
}
```

**Comment by bwilkerson** [comment]
Kind of sounds like something that ought to be caught by `unawaited_futures`.

**Comment by shyndman** [comment]
It certainly does. Good idea.
